src/app/ReadClient.cpp

```diff
@@ -894,32 +917,58 @@ void ReadClient::OnLivenessTimeoutCallback(System::Layer * apSystemLayer, void *
+        // This decision process is triggered by either a liveness timeout or the arrival of a check-in message.
+        // In the case of a liveness timeout, CHIP_ERROR_LIT_SUBSCRIBE_INACTIVE_TIMEOUT is passed to application and indicate LIT
+        // ICD device is inactive. In the case of a check-in message, the `OnActiveModeNotification` callback is invoked and
+        // the `InactiveICDSubscription` state is specifically checked. Note that the liveness timeout duration always exceeds
+        // the maximum interval time (and hence the idle duration). Therefore, transitioning the device to `InactiveICDSubscription`
+        // upon reaching the liveness timeout is safe and consistent.
+        ChipLogProgress(DataManagement, "Peer is not active now, mark the subscription as InactiveICDSubscription.");
+        _this->MoveToState(ClientState::InactiveICDSubscription);
+
+        // Mark defunct immediately as usual flow does
+        _this->TriggerResubscriptionForLivenessTimeout(CHIP_ERROR_LIT_SUBSCRIBE_INACTIVE_TIMEOUT);
+
+        // Return early here since the peer is idle and cannot receive any messages. The resubscription will be deferred until
+        // `OnActiveModeNotification` is invoked.
+        return;
+    }
+
+    _this->TriggerResubscriptionForLivenessTimeout(CHIP_ERROR_TIMEOUT);
+}
+
+void ReadClient::TriggerResubscriptionForLivenessTimeout(CHIP_ERROR aReason)
+{
     // We didn't get a message from the server on time; it's possible that it no
     // longer has a useful CASE session to us.  Mark defunct all sessions that
     // have not seen peer activity in at least as long as our session.
-    const auto & holder = _this->mReadPrepareParams.mSessionHolder;
+    const auto & holder = mReadPrepareParams.mSessionHolder;
     if (holder)
     {
         System::Clock::Timestamp lastPeerActivity = holder->AsSecureSession()->GetLastPeerActivityTime();
-        _this->mpImEngine->GetExchangeManager()->GetSessionManager()->ForEachMatchingSession(
-            _this->mPeer, [&lastPeerActivity](auto * session) {
-                if (!session->IsCASESession())
-                {
-                    return;
-                }
+        mpImEngine->GetExchangeManager()->GetSessionManager()->ForEachMatchingSession(mPeer, [&lastPeerActivity](auto * session) {
+            if (!session->IsCASESession())
+            {
+                return;
+            }
 
-                if (session->GetLastPeerActivityTime() > lastPeerActivity)
-                {
-                    return;
-                }
+            if (session->GetLastPeerActivityTime() > lastPeerActivity)
+            {
+                return;
+            }
 
-                session->MarkAsDefunct();
-            });
+            session->MarkAsDefunct();
+        });
     }
 
     // TODO: add a more specific error here for liveness timeout failure to distinguish between other classes of timeouts (i.e
     // response timeouts).
```

Comment:
This comment does not seem to be in the right place now...  It should be over where we pass CHIP_ERROR_TIMEOUT, no?

Reply:
Moved